### ISSUE TITLE
Linux, "bad address" while setting DTR

### ISSUE THREAD

**Issue by milabs** [albenik/go-serial#41]
Hello,

There is a problem with `DTR` line setting while running on Linux:

~~~
package main

import (
	"log"
	"github.com/albenik/go-serial/v2"
)

func main() {
	port, err := serial.Open("/dev/ttyUSB0",
		serial.WithBaudrate(115200),
		serial.WithDataBits(8),
		serial.WithParity(serial.NoParity),
		serial.WithStopBits(serial.OneStopBit),
	)

	if err != nil {
		log.Fatal(err)
	}

	defer port.Close()

	log.Println("Opened", port)

	if err := port.SetDTR(true); err != nil {
		log.Println(err)
	}
}
~~~

Running as:

~~~
$ go run test.go
2024/01/24 23:00:11 Opened /dev/ttyUSB0
2024/01/24 23:00:11 operating system error: bad address
~~~

Device (converter) information from `dmesg`:

~~~
[1113444.099145] usb 1-1: new full-speed USB device number 5 using xhci_hcd
[1113444.226199] usb 1-1: New USB device found, idVendor=1a86, idProduct=7523, bcdDevice= 2.64
[1113444.226213] usb 1-1: New USB device strings: Mfr=0, Product=2, SerialNumber=0
[1113444.226220] usb 1-1: Product: USB Serial
[1113444.228034] ch341 1-1:1.0: ch341-uart converter detected
[1113444.228747] usb 1-1: ch341-uart converter now attached to ttyUSB0
~~~

**Comment by milabs** [comment]
Ok, I've found the problem:

https://github.com/albenik/go-serial/blob/9b94bc9eff70f16375742aada988811234593f28/serial_unix.go#L384-L389

Should use `IoctlSetPointerInt` instead of `IoctlSetInt` =)

cc: @albenik

**Comment by albenik** [comment]
@milabs Thanks! Merged & tagged `v2.6.1`
